tests/Unit/Schema/Resolver/ResolverRegistryTest.php

```diff
@@ -205,3 +232,13 @@ public function resolve($rootValue, array $arguments, $context = null, ?ResolveI
         return \sprintf('Hello %s!', $arguments['name'] ?? 'world');
     }
 }
+
+class HelloResolverWithSpecificMiddleware extends HelloResolver
+{
+    public function getMiddleware(): ?array
+    {
+        return [
+            'logInput',
```

Comment:
We could use the FQCN as well.

Reply:
I personally prefer FQCN over stringly typed short names.

Reply:
Agreed, this is very JavaScripty now that I look at it.

Reply:
I’ll change this in a moment.

Reply:
Same when registering them. `SomeClass::class` is the safest option, and even easier to type if you have auto-completion

Reply:
@spawnia I would prefer using the `new` keyword for registering them because this allows you to do DI if necessary. Or would you prefer the option of using both the `new` keyword and FQCN for both resolvers and middleware?

Reply:
The way you put it now is perfect. Eliminating the need to come up with a short name for them is better.